pkg/river/internal/stdlib/stdlib.go

```diff
@@ -116,4 +118,15 @@ var Identifiers = map[string]interface{}{
 
 		return args[len(args)-1], nil
 	}),
+
+	"format":      fmt.Sprintf,
+	"join":        strings.Join,
```

Comment:
If we reverse the arguments can we do `join("one","two",",")`? And ignore the need of forcing an array?

Reply:
I also wonder about `join` has the name since that is pretty generic. IE by name join and concat seem really similar.

Reply:
> If we reverse the arguments can we do join("one","two",",")? And ignore the need of forcing an array?

IMO: outside of format (where there's no choice), I'd prefer to avoid functions that take variable length arguments where different arguments hold different meanings, because I'm worried that it makes it easier to get wrong, and I'm worried it's harder to document.

Reply:
I also think in the case of `join`, it's less useful to pass a list of array elements manually. You are more likely to have an array value from another component that you're trying to join on. By making it take variable arugments, there would be no way of passing an array to it (short of adding a Go-style `...` splat operator, which I'm not convinced we're ready for yet.)

Reply:
From my point of view, the function `join` is responsible for convert a list a of string into a string with a specific separator. I do not know much languages, but in each language, it always same. I would recommend to keep this pattern in river, too.

Reply:
Do we have an agreement on the current implement of the `join` function?

Reply:
Whats the behavior if you pass in non string elements?

Reply:
> join([{a="b",}, 0, 1], ", ")

returns

> {a = "b"} should be string, got object